### PR TITLE
Mic 4277/update 1040 obs

### PR DESCRIPTION
**Pull request by albrja**
### Updates 1040 observer to fix sampling bugs.
- *Category*: Bugfix
- *JIRA issue*: [MIC-4277](https://jira.ihme.washington.edu/browse/MIC-4277)
- *Research reference*: 

### Changes and notes
-Adds sampling to replace income sampling at flat rate
-Fixes bug with joint filing eligibility.

### Verification and Testing
Successfully ran simulation.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,3 +1,8 @@
+**1.5.0 - 07/27/23**
```
Comment (stevebachmeier) [inline CHANGELOG.rst:1]: nit: this feels more like a patch release to me

File: src/vivarium_census_prl_synth_pop/components/observers.py [hunk 2]
```diff
@@ -1064,8 +1072,13 @@ def get_observation(self, event: Event) -> pd.DataFrame:
             "Opp-sex spouse",
             "Same-sex spouse",
         ]
+        # Get partners of reference members who filed taxes
+        hh_ids_with_filing_ref_persons = pop.loc[
+            pop["relationship_to_reference_person"] == "Reference person", "household_id"
+        ]
         partners_of_household_head_idx = pop.index[
```
Comment (stevebachmeier) [inline src/vivarium_census_prl_synth_pop/components/observers.py:1079]: nit: partners_of_reference_person_idx

**Review by stevebachmeier — APPROVED** [review]
I assume you ran `pytest` on this? Remember we are NOT running during build for this repo.

Also, do you think it's worth writing a unit test for this bugfix? I kinda feel like every bugfix should have a unit test written for it, but that also seems like a lot of potentially not-that-useful work.

**Review by rmudambi — APPROVED** [review]

**Comment by albrja** [comment]
@stevebachmeier Yes I ran test suites. I think the unit test is a good idea but I'm hesitant right now because at some point we will add income as our sampling factor for who files taxes. I'm unsure where we will do that - whether is in this same spot or if we can filter later in pseudopeople.

**Pull request merged**